ext/opentelemetry-ext-http-requests/src/opentelemetry/ext/http_requests/__init__.py

```diff
@@ -65,7 +65,7 @@ def instrumented_request(self, method, url, *args, **kwargs):
                 path = "<URL parses to None>"
             path = parsed_url.path
 
-        with tracer.start_span(path, kind=SpanKind.CLIENT) as span:
+        with tracer.start_as_current_span(path, kind=SpanKind.CLIENT) as span:
```

Comment:
This package doesn't need/use current span reference, feels like start_span is actually available for this kind of scenarios

Reply:
That's a good point, maybe we do want `start_span` here.

We may want to put spans into the context *by default*, even if we close the span before returning control. If `start_span` started and activated the span, and we had another method named `start_span_but_dont_activate`, would it still be a good idea to use the latter here?

@Oberon00 what do you think?

Reply:
the current span is actually referenced here, abstracted in the propagators layer:

https://github.com/open-telemetry/opentelemetry-python/blob/30568defe7a8a17a52e17ac5056162e6c2699a31/ext/opentelemetry-ext-http-requests/src/opentelemetry/ext/http_requests/__init__.py#L77

https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-api/src/opentelemetry/propagators/__init__.py#L47

Which I think is the correct behavior. I'm sure others will also rely on the assumption that the span was started here, regardless of whether the integration directly uses the context or not.

I would argue that most common cases require the tracer to use the new span, and we should only opt out of that behavior in advanced cases.

I'm +1 for start_span_but_dont_activate, or maybe pass that it as a keyword argument. (will_activate=False).

Reply:
I think this is either a misusage of the inject API or the inject API should provide a way to supply the Span to propagate.

But I definitely agree on using `start_as_current_span` here. E.g. consider what should happen if we decided to represent HTTP redirects as their own spans: They would need to become child spans.

Reply:
In general, I view the active span more as an inter-instrumentation communication. If only a single instrumentation library was involved it could devise its own mechanism to manage parent spans easily (although ofc relying on the current span should still be preferred for parent/child relations).